### PR TITLE
DOC: update pandas.core.resample.Resampler.nearest docstring

### PR DESCRIPTION
**Pull request by thicorfon**
Checklist for the pandas documentation sprint:

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py pandas.core.resample.Resampler.nearest`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single pandas.core.resample.Resampler.nearest`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
############## Docstring (pandas.core.resample.Resampler.nearest) ##############
################################################################################

Fill the new missing values with their nearest neighbor value, based
on index.

When resampling data, missing values may appear (e.g., when the
resampling frequency is higher than the original frequency).
The nearest fill will replace ``NaN`` values that appeared in
the resampled data with the value from the nearest member of the
sequence, based on the index value.
Missing values that existed in the original data will not be modified.
If `limit` is given, fill only `limit` values in each direction for
each of the original values.

Parameters
----------
limit : integer, optional
    Limit of how many values to fill.

    .. versionadded:: 0.21.0

Returns
-------
Series, DataFrame
    An upsampled Series or DataFrame with ``NaN`` values filled with
    their closest neighbor value.

See Also
--------
backfill: Backward fill the new missing values in the resampled data.
fillna : Fill ``NaN`` values using the specified method, which can be
    'backfill'.
pad : Forward fill ``NaN`` values.
pandas.Series.fillna : Fill ``NaN`` values in the Series using the
    specified method, which can be 'backfill'.
pandas.DataFrame.fillna : Fill ``NaN`` values in the DataFrame using
    the specified method, which can be 'backfill'.

Examples
--------

Resampling a Series:

>>> s = pd.Series([1, 2, 3],
...               index=pd.date_range('20180101', periods=3,
...                                   freq='1h'))
>>> s
2018-01-01 00:00:00    1
2018-01-01 01:00:00    2
2018-01-01 02:00:00    3
Freq: H, dtype: int64

>>> s.resample('20min').nearest()
2018-01-01 00:00:00    1
2018-01-01 00:20:00    1
2018-01-01 00:40:00    2
2018-01-01 01:00:00    2
2018-01-01 01:20:00    2
2018-01-01 01:40:00    3
2018-01-01 02:00:00    3
Freq: 20T, dtype: int64

Resample in the middle:

>>> s.resample('30min').nearest()
2018-01-01 00:00:00    1
2018-01-01 00:30:00    2
2018-01-01 01:00:00    2
2018-01-01 01:30:00    3
2018-01-01 02:00:00    3
Freq: 30T, dtype: int64

Limited fill:

>>> s.resample('10min').nearest(limit=1)
2018-01-01 00:00:00    1.0
2018-01-01 00:10:00    1.0
2018-01-01 00:20:00    NaN
2018-01-01 00:30:00    NaN
2018-01-01 00:40:00    NaN
2018-01-01 00:50:00    2.0
2018-01-01 01:00:00    2.0
2018-01-01 01:10:00    2.0
2018-01-01 01:20:00    NaN
2018-01-01 01:30:00    NaN
2018-01-01 01:40:00    NaN
2018-01-01 01:50:00    3.0
2018-01-01 02:00:00    3.0
Freq: 10T, dtype: float64

Resampling a DataFrame that has missing values:

>>> df = pd.DataFrame({'a': [2, np.nan, 6], 'b': [1, 3, 5]},
...                   index=pd.date_range('20180101', periods=3,
...                                       freq='h'))
>>> df
                       a  b
2018-01-01 00:00:00  2.0  1
2018-01-01 01:00:00  NaN  3
2018-01-01 02:00:00  6.0  5

>>> df.resample('20min').nearest()
                       a  b
2018-01-01 00:00:00  2.0  1
2018-01-01 00:20:00  2.0  1
2018-01-01 00:40:00  NaN  3
2018-01-01 01:00:00  NaN  3
2018-01-01 01:20:00  NaN  3
2018-01-01 01:40:00  6.0  5
2018-01-01 02:00:00  6.0  5

Resampling a DataFrame with shuffled indexes:

>>> df = pd.DataFrame({'a': [2, 6, 4]},
...                   index=pd.date_range('20180101', periods=3,
...                                       freq='h'))
>>> df
                     a
2018-01-01 00:00:00  2
2018-01-01 01:00:00  6
2018-01-01 02:00:00  4

>>> sorted_df = df.sort_values(by=['a'])
>>> sorted_df
                     a
2018-01-01 00:00:00  2
2018-01-01 02:00:00  4
2018-01-01 01:00:00  6

>>> sorted_df.resample('20min').nearest()
                     a
2018-01-01 00:00:00  2
2018-01-01 00:20:00  2
2018-01-01 00:40:00  6
2018-01-01 01:00:00  6
2018-01-01 01:20:00  6
2018-01-01 01:40:00  4
2018-01-01 02:00:00  4

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Errors in parameters section
		Parameter "limit" description should finish with "."
```

The error is caused by `.. versionadded:: 0.21.0`, which has no period in the end.

### CODE REVIEW

File: pandas/core/resample.py [hunk 1]
```diff
@@ -498,23 +498,142 @@ def pad(self, limit=None):
 
     def nearest(self, limit=None):
         """
-        Fill values with nearest neighbor starting from center
+        Fill the new missing values with their nearest neighbor value, based
```
Comment (TomAugspurger) [inline pandas/core/resample.py]: This should fit on a single line. Can you try rephrasing?

File: pandas/core/resample.py [hunk 2]
```diff
@@ -498,23 +498,142 @@ def pad(self, limit=None):
...
+        2018-01-01 00:00:00    1
+        2018-01-01 00:20:00    1
+        2018-01-01 00:40:00    2
+        2018-01-01 01:00:00    2
+        2018-01-01 01:20:00    2
+        2018-01-01 01:40:00    3
+        2018-01-01 02:00:00    3
+        Freq: 20T, dtype: int64
+
+        Resample in the middle:
```
Comment (TomAugspurger) [inline pandas/core/resample.py]: What do you mean by "in the middle?"

File: pandas/core/resample.py [hunk 3]
```diff
@@ -498,23 +498,142 @@ def pad(self, limit=None):
...
+        2018-01-01 00:00:00    1
+        2018-01-01 00:30:00    2
+        2018-01-01 01:00:00    2
+        2018-01-01 01:30:00    3
+        2018-01-01 02:00:00    3
+        Freq: 30T, dtype: int64
+
+        Limited fill:
+
+        >>> s.resample('10min').nearest(limit=1)
```
Comment (TomAugspurger) [inline pandas/core/resample.py]: This is a tad long. Can you change it to `'20min'`. I think that'll still make the point as the first will be filled and the second will be NaN

File: pandas/core/resample.py (hunk 1 above)
Comment (jacko8944) [reply pandas/core/resample.py]: Lundy06032013@gmail.com

Comment (jacko8944) [reply pandas/core/resample.py]: Lundy06032013@gmail.com

File: pandas/core/resample.py [hunk 4]
```diff
@@ -498,23 +498,142 @@ def pad(self, limit=None):
...
+                               a  b
+        2018-01-01 00:00:00  2.0  1
+        2018-01-01 00:20:00  2.0  1
+        2018-01-01 00:40:00  NaN  3
+        2018-01-01 01:00:00  NaN  3
+        2018-01-01 01:20:00  NaN  3
+        2018-01-01 01:40:00  6.0  5
+        2018-01-01 02:00:00  6.0  5
+
+        Resampling a DataFrame with shuffled indexes:
```
Comment (jorisvandenbossche) [inline pandas/core/resample.py]: I am not fully sure this example is needed, as it is something general to `resample`, and not specific to the  `nearest` method

File: pandas/core/resample.py [hunk 5]
```diff
@@ -498,23 +498,142 @@ def pad(self, limit=None):
...
+        2018-01-01 01:00:00    2.0
+        2018-01-01 01:10:00    2.0
+        2018-01-01 01:20:00    NaN
+        2018-01-01 01:30:00    NaN
+        2018-01-01 01:40:00    NaN
+        2018-01-01 01:50:00    3.0
+        2018-01-01 02:00:00    3.0
+        Freq: 10T, dtype: float64
+
+        Resampling a DataFrame that has missing values:
```
Comment (jorisvandenbossche) [inline pandas/core/resample.py]: I would add as small explanation about that the initial NaN is preserved

**Comment by jreback** [comment]
@datapythonista

**Review by datapythonista — APPROVED** [review]
Rebased, added couple of minor fixes, and made the examples shorter.

@jbrockmendel if you don't mind taking a look here too and merging on green if looks good. I'm trying to close all pending PRs from the docsting. I'm happy if what we merge if correct, future improvements can be addressed later in separate PRs. Thanks!

File: pandas/core/resample.py [hunk 6]
```diff
@@ -418,23 +418,80 @@ def pad(self, limit=None):
 
     def nearest(self, limit=None):
         """
-        Fill values with nearest neighbor starting from center
+        Resample by using the nearest value.
+
+        When resampling data, missing values may appear (e.g., when the
+        resampling frequency is higher than the original frequency).
+        The nearest fill will replace ``NaN`` values that appeared in
```
Comment (jbrockmendel) [inline pandas/core/resample.py]: "nearest fill" refers to this method, right?  Maybe make that explicit with "`nearest` fill method"?

File: pandas/core/resample.py [hunk 7]
```diff
@@ -418,23 +418,80 @@ def pad(self, limit=None):
...
+        the resampled data with the value from the nearest member of the
+        sequence, based on the index value.
+        Missing values that existed in the original data will not be modified.
+        If `limit` is given, fill only `limit` values in each direction for
```
Comment (jbrockmendel) [inline pandas/core/resample.py]: Maybe the second "`limit`" could be "this many"?  Probably worthwhile to get a non-native speaker to weight in what is clearest.

File: pandas/core/resample.py [hunk 8]
```diff
@@ -418,23 +418,80 @@ def pad(self, limit=None):
...
-            limit of how many values to fill
+        limit : int, optional
+            Limit of how many values to fill.
 
             .. versionadded:: 0.21.0
 
         Returns
         -------
-        an upsampled Series
+        Series or DataFrame
```
Comment (jbrockmendel) [inline pandas/core/resample.py]: @datapythonista do we have a convention for saying same-type-as-input?

File: pandas/core/resample.py [hunk 9]
```diff
@@ -418,23 +418,80 @@ def pad(self, limit=None):
...
         --------
-        Series.fillna
-        DataFrame.fillna
+        backfill: Backward fill the new missing values in the resampled data.
+        fillna : Fill ``NaN`` values using the specified method, which can be
+            'backfill'.
+        pad : Forward fill ``NaN`` values.
+        pandas.Series.fillna : Fill ``NaN`` values in the Series using the
+            specified method, which can be 'backfill'.
+        pandas.DataFrame.fillna : Fill ``NaN`` values in the DataFrame using
```
Comment (jbrockmendel) [inline pandas/core/resample.py]: The thoroughness is good, but this seems excessive.  @datapythonista what's the convention for how much to write in this section?

File: pandas/core/resample.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/resample.py]: +1

**Comment by datapythonista** [comment]
Thanks for another great review @jbrockmendel. Made the changes, I think this should be ready to be merged on green.

For the "same type as caller", we try to use only Python types in the types of the return and parameters. Ideally at some point we can parse those types and detect suspicious things or get statistics. In generic methods, so far we're using `Series or DataFrame`. At a later stage it'd be nice to write the corresponding class, but that requires a bit of thinking on how to do, and will come in a separate PR.

**Comment by datapythonista** [comment]
@jreback this should be ready to merge, if you want to have a look

**Comment by jreback** [comment]
thanks @thicorfon

**Pull request merged**